### PR TITLE
Change DeliverTo from a Coproduct to a sealed trait + case classes

### PR DESCRIPTION
**Pull request by cb372**
This requires upgrading to a currently-unreleased version of avro4s. I've forked avro4s and released the latest master as `1.6.4-ovo-1` on Bintray.

Also: 
* Remove the last traces of enumeratum, to fix the dreaded knownDirectSubclasses compile error
* Upgrade to the latest comms-kafka-serialisation, which uses avro4s instead of circe for deserialisation

### CODE REVIEW

File: src/main/scala/com/ovoenergy/comms/model/DeliverTo.scala [hunk 1]
```diff
@@ -1,35 +1,17 @@
...
-  def getCustomerId(deliverTo: DeliverTo): Option[CustomerId] = deliverTo.value.select[CustomerId]
-
-  def getContactDetails(deliverTo: DeliverTo): Option[ContactDetails] = deliverTo.value.select[ContactDetails]
+sealed trait NonCustomer extends DeliverTo {
+  override def mdcMap = Map.empty
 }
+
+case class EmailAddressOnly(emailAddress: String)                                 extends NonCustomer
+case class MobileNumberOnly(phoneNumber: String)                                  extends NonCustomer
+case class EmailAddressAndMobileNumber(emailAddress: String, phoneNumber: String) extends NonCustomer
```
Comment (fleetwoods) [inline src/main/scala/com/ovoenergy/comms/model/DeliverTo.scala]: Is this scalable? Add in Address, Push Token, etc and combinations..does having a single case class with Options work better?

File: build.sbt [hunk 2]
```diff
@@ -17,13 +17,13 @@ testOptions in Test += Tests.Argument(TestFrameworks.ScalaCheck,
 mappings in (Compile, packageBin) ~= { _.filter { case (absPath, logicalPath) => logicalPath.startsWith("com/ovoenergy/comms") } }
 resolvers += Resolver.bintrayRepo("ovotech", "maven")
 libraryDependencies ++= Seq(
-  "com.sksamuel.avro4s" %% "avro4s-core" % "1.6.2",
+  "com.sksamuel.avro4s" %% "avro4s-core" % "1.6.5-SNAPSHOT",
```
Comment (fleetwoods) [inline build.sbt]: We shouldn't merge with a snapshot as may well change on us, would you fork and release to our repo or something?

File: src/main/scala/com/ovoenergy/comms/model/DeliverTo.scala (hunk 1 above)
Comment (cb372) [inline src/main/scala/com/ovoenergy/comms/model/DeliverTo.scala]: yes, happy to put it back to how it was, I was just working around a problem where Avro union types can't have union types as direct children

File: build.sbt (hunk 2 above)
Comment (cb372) [inline build.sbt]: yeah I guess we'd fork it and release it to bintray as `1.6.4-ovo` or something like that, just until 1.6.5 is released

Comment (fleetwoods) [inline build.sbt]: 👍

File: build.sbt [hunk 3]
```diff
@@ -17,13 +17,12 @@ testOptions in Test += Tests.Argument(TestFrameworks.ScalaCheck,
...
 libraryDependencies ++= Seq(
-  "com.sksamuel.avro4s" %% "avro4s-core" % "1.6.2",
+  "com.sksamuel.avro4s" %% "avro4s-core" % "1.6.4-ovo-1",
   "org.typelevel" %% "cats-core" % "0.9.0",
-  "com.beachape" %% "enumeratum" % "1.5.6",
   "org.scalacheck" %% "scalacheck" % "1.13.4" % Test,
   "com.github.alexarchambault" %% "scalacheck-shapeless_1.13" % "1.1.4" % Test,
-  "com.ovoenergy" %% "comms-kafka-serialisation" % "2.2" % Test,
-  "io.circe" %% "circe-generic" % circeVersion % Test
+  "com.ovoenergy" %% "comms-kafka-serialisation" % "2.3-SNAPSHOT" % Test,
```
Comment (cb372) [inline build.sbt]: TODO bump version when released

File: src/main/scala/com/ovoenergy/comms/model/ErrorCode.scala [hunk 4]
```diff
@@ -24,21 +36,22 @@ object ErrorCode extends Enum[ErrorCode] {
     override def apply(value: ErrorCode) = value.toString
   }
 
+  // TODO some shapeless magic to replace this boilerplate
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/comms/model/ErrorCode.scala]: I'll try this in a separate PR. Not important right now.

**Pull request merged**